File: 03 Podminky a cykly/06 uzaver.ts
function tovarnaNaScitaciFunkce(a: number) {
  return function pricti(b: number) {
    return a + b;
  };
}

const prictiDva = tovarnaNaScitaciFunkce(2);
console.log(prictiDva(8));

function tovarnaNaNasobiciFunkce(a: number) {
  return (b: number) => a * b;
}

const vynasobPulkou = tovarnaNaNasobiciFunkce(0.5);
console.log(vynasobPulkou(8));

function tovarnaNaFunkceProdukujiciId() {
  let currentId = 0;
  return () => currentId++;
}

const generujId = tovarnaNaFunkceProdukujiciId();
console.log(generujId());
console.log(generujId());
console.log(generujId());
console.log(generujId());
console.log(generujId());

File: 03 Podminky a cykly/06 uzaver.py
def tovarnaNaScitaciFunkce(a: int):
    def pricti(b: int):
        return a + b
    return pricti


prictiDva = tovarnaNaScitaciFunkce(2)
print(prictiDva(8))


def tovarnaNaNasobiciFunkce(a):
    return lambda b: a * b


vynasobPulkou = tovarnaNaNasobiciFunkce(0.5)
print(vynasobPulkou(8))


def tovarnaNaFunkceProdukujiciId():
    currentId = 0

    def vratNoveId():
        nonlocal currentId
        currentId += 1
        return currentId
    return vratNoveId


generujId = tovarnaNaFunkceProdukujiciId()
print(generujId())
print(generujId())
print(generujId())
print(generujId())
print(generujId())
